Source/Diagnostics/ParticleIO.cpp

```diff
@@ -103,7 +105,84 @@ void
 MultiParticleContainer::Restart (const std::string& dir)
 {
     for (unsigned i = 0, n = species_names.size(); i < n; ++i) {
-        allcontainers[i]->Restart(dir, species_names[i]);
+        WarpXParticleContainer* pc = allcontainers[i].get();
+        std::string header_fn = dir + "/" + species_names[i] + "/Header";
+
+        Vector<char> fileCharPtr;
+        ParallelDescriptor::ReadAndBcastFile(header_fn, fileCharPtr);
+        std::string fileCharPtrString(fileCharPtr.dataPtr());
+        std::istringstream is(fileCharPtrString, std::istringstream::in);
```

Comment:
Avoiding silent read corruptions:
```suggestion
        std::istringstream is(fileCharPtrString, std::istringstream::in);
        is.exceptions(std::ios_base::failbit | std::ios_base::badbit);
```